allauth/account/adapter.py

```diff
@@ -708,6 +708,9 @@ def get_client_ip(self, request):
             ip = request.META.get("REMOTE_ADDR")
         return ip
 
+    def get_http_user_agent(self, request):
+        return request.META["HTTP_USER_AGENT"]
```

Comment:
The spec says that a browser "should" send the user agent. So we should not crash if the header is not there. I would suggest:

```suggestion
        return request.META.get("HTTP_USER_AGENT",  "Unspecified")
```

This way, you can also leave out sending the UA in the various test cases.

Reply:
Noted